bin/dep

```diff
@@ -1,29 +1,40 @@
-#!/bin/sh
+#!/bin/bash
+# bash is required since indirect variable substitution is used.
 
 set -eu
 
+# Keep this in sync with Dockerfile-go-deps. The digests will be different for each
+# version and each platform; they can be found in the *.sha256 files alongside the
+# executables at ${dep_base_url}.
+depversion=0.4.1
+dep_base_url="https://github.com/golang/dep/releases/download/v${depversion}/"
+dep_digest_linux=31144e465e52ffbc0035248a10ddea61a09bf28b00784fd3fdd9882c8cbb2315
+dep_digest_darwin=f170008e2bf8b196779c361a4eaece1b03450d23bbf32d1a0beaa9b00b6a5ab4
+dep_digest_windows=f6e6a872c54d5ae7536ac71fd5bcac9f4e7b8a1dafa1ef7c23866e2f3069fe4e
+
 cd "$(pwd -P)"
 
+os=linux
+exe=
+check_sha256="sha256sum -c --quiet"
 if [ "$(uname -s)" = "Darwin" ]; then
   os=darwin
-else
-  os=linux
-fi
-if [ "$(uname -m)" = "x86_64" ]; then
-  arch=386
-else 
-  arch=amd64
+  check_sha256="shasum -a 256 -p"
+elif [ "$(uname -o)" = "Msys" ]; then
+  os=windows
+  exe=.exe
 fi
 
-depbin=.dep
-depversion=0.4.1 # Need to keep this in sync with Dockerfile-go-deps
-depurl="https://github.com/golang/dep/releases/download/v${depversion}/dep-${os}-${arch}"
+depbin=.dep${exe}
+depurl="${dep_base_url}dep-${os}-amd64${exe}"
 
 if [ ! -f "$depbin" ]; then
   tmp=$(mktemp -d -t dep.XXX)
   (
     cd "$tmp"
     curl -L --silent --fail -o "$depbin" "$depurl"
+    ddv="dep_digest_${os}"
+    echo ${!ddv} $depbin | ${check_sha256}
```

Comment:
it looks like we compute the SHA but don't actually check it against the expected SHA?

Reply:
on darwin this worked:
```bash
echo ${!ddv} ?$depbin | ${check_sha256} -c -
```

Reply:
@adleong Ooops! I was forgetting to pass `-c` to `shasum` like I did for sha256sum.

@siggy Thanks. That's similar to what I've done now, except I use "*" instead of "?". Could you check that the new version is working.

For some reason I had previously thought MinGW was missing `shasum`; however, it actually has it. Thus, we should be able to use it on all platforms. I've updated the PR to do so.